lib/createExplorer.js

```diff
@@ -10,10 +10,10 @@ var funcRunner = require('./funcRunner');
 
 module.exports = function (options) {
   // When `options.sync` is `false` (default),
-// these cache Promises that resolve with results, not the results themselves.
+  // these cache Promises that resolve with results, not the results themselves.
   var fileCache = (options.cache) ? new Map() : null;
   var directoryCache = (options.cache) ? new Map() : null;
-  var transform = options.transform || (options.sync) ? identitySync : identityPromise;
+  var transform = options.transform || (options.sync ? identitySync : identityPromise);
```

Comment:
I was surprised too. I completely agree with the suggested change. On a side note, there is actually no need to have a different identity func for sync and async. We can use the sync version for both.